### PR TITLE
Add spruce recipe

### PR DESCRIPTION
**Pull request by asford**
Checklist

- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/spruce/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    url: https://{{ goname }}/archive/v{{ version }}.tar.gz
+    sha256: 39278f423c979688b14817fe48fc80fb7bd828fd864ded50edede302342aab60
+
+build:
+  number: 0
+  script:
+    - GOBIN=$(go env GOBIN)  # [not win]
+    - FOR /F "delims=" %%i IN ('go env GOBIN') DO set "GOBIN=%%i"  # [win]
+    - pushd {{ pkg_src }}
+    # TODO build tag?
```
Comment (asford) [inline recipes/spruce/meta.yaml:22]:
```suggestion
```

File: recipes/spruce/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    url: https://{{ goname }}/archive/v{{ version }}.tar.gz
+    sha256: 39278f423c979688b14817fe48fc80fb7bd828fd864ded50edede302342aab60
+
+build:
+  number: 0
+  script:
+    - GOBIN=$(go env GOBIN)  # [not win]
+    - FOR /F "delims=" %%i IN ('go env GOBIN') DO set "GOBIN=%%i"  # [win]
+    - pushd {{ pkg_src }}
+    - go build -v -o $GOBIN/{{ name }}  -ldflags "-X main.Version={{ version }}" ./cmd/spruce# [unix]
```
Comment (asford) [inline recipes/spruce/meta.yaml:22]: /
```suggestion
    - go build -v -o $GOBIN/{{ name }}  -ldflags "-X main.Version={{ version }}" ./cmd/spruce  # [unix]
```

**Comment by asford** [comment]
@conda-forge/help-go This is ready for review, thanks!

File: recipes/spruce/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - FOR /F "delims=" %%i IN ('go env GOBIN') DO set "GOBIN=%%i"  # [win]
+    - pushd {{ pkg_src }}
+    - go build -v -o $GOBIN/{{ name }}  -ldflags "-X main.Version={{ version }}" ./cmd/spruce  # [unix]
+    - go build -v -o %GOBIN%\{{ name }}.exe -ldflags "-X main.Version={{ version }}" .\cmd\spruce  # [win]
+    - go-licenses save ./cmd/spruce --save_path=./license-files  # [unix]
+    - go-licenses save .\cmd\spruce --save_path=.\license-files  # [win]
+
+requirements:
+  build:
+    - {{ compiler('go') }}
```
Comment (xhochy) [inline recipes/spruce/meta.yaml:29]:
```suggestion
    - {{ compiler('go_nocgo') }}
```

File: recipes/spruce/conda_build_config.yaml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+go_compiler:
```
Comment (xhochy) [inline recipes/spruce/conda_build_config.yaml:1]: I would prefer that you use the compiler directly below and not have a cbc.yaml.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/spruce/conda_build_config.yaml (hunk 4 above)
Comment (asford) [reply recipes/spruce/conda_build_config.yaml]: Perfect, thanks!

File: recipes/spruce/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    - FOR /F "delims=" %%i IN ('go env GOBIN') DO set "GOBIN=%%i"  # [win]
+    - pushd {{ pkg_src }}
+    - go build -v -o $GOBIN/{{ name }}  -ldflags "-X main.Version={{ version }}" ./cmd/spruce  # [unix]
+    - go build -v -o %GOBIN%\{{ name }}.exe -ldflags "-X main.Version={{ version }}" .\cmd\spruce  # [win]
+    - go-licenses save ./cmd/spruce --save_path=./license-files  # [unix]
+    - go-licenses save .\cmd\spruce --save_path=.\license-files  # [win]
+
+requirements:
+  build:
+    - {{ compiler('go_nocgo') }}
```
Comment (xhochy) [inline recipes/spruce/meta.yaml:29]:
```suggestion
    - {{ compiler('go-nocgo') }}
```

**Review by xhochy — APPROVED** [review]

**Pull request merged**